test/Inputs/clang-importer-sdk/usr/include/objc_structs.h

```diff
@@ -8,10 +8,19 @@ struct StructOfBlocks {
   void (^__unsafe_unretained _Nonnull block)(void);
 };
 
-struct StrongsInAStruct { // expected-note {{record 'StrongsInAStruct' is not trivial to copy/destroy}}
-  __strong NSString *nsstr;
+@interface MYObject: NSObject
+@end
+
+struct StrongsInAStruct {
+  __strong MYObject *_Nonnull myobj;
+};
+
+struct WeaksInAStruct {
+  // Weak pointers are always _Nullable
+  __weak MYObject *_Nullable myobj;
 };
 
-struct WeaksInAStruct { // expected-note {{record 'WeaksInAStruct' is not trivial to copy/destroy}}
-  __weak NSString *nsstr;
+struct WeakAndNonnull {
+  // Weak pointers cannnnn be annotated _Nonnull
+  __weak MYObject *_Nonnull myobj;
```

Comment:
Ah I see, I really though this wasn't allowed

Reply:
On clang totally fine - Swift won't be able to represent an interface for this

Reply:
Yes, I just asked because at property level if we try `@property (weak, nonnull) NSString* prop;` we get `property attributes 'nonnull' and 'weak' are mutually exclusive` compiler error. So that is why I thought this wasn't allowed.